File: service/src/main/java/hu/bsstudio/robonaut/team/mapper/ScoreEntityMapper.java
package hu.bsstudio.robonaut.team.mapper;

import hu.bsstudio.robonaut.entity.ScoreEntity;
import hu.bsstudio.robonaut.team.model.Score;

public class ScoreEntityMapper {

    public ScoreEntity toEntity(final Score score) {
        final var entity = new ScoreEntity();
        entity.setScore(score.getTotalScore());
        entity.setSpeedScore(score.getSpeedScore());
        entity.setBestSpeedTime(score.getBestSpeedTime());
        return entity;
    }

    public Score toModel(final ScoreEntity entity) {
        return new Score(entity.getSpeedScore(), entity.getBestSpeedTime(), entity.getScore());
    }
}

File: service/src/main/kotlin/hu/bsstudio/robonaut/team/mapper/ScoreEntityMapper.kt
package hu.bsstudio.robonaut.team.mapper

import hu.bsstudio.robonaut.entity.ScoreEntity
import hu.bsstudio.robonaut.team.model.Score

class ScoreEntityMapper {

    fun toEntity(score: Score) = ScoreEntity(
        score = score.totalScore,
        speedScore = score.speedScore,
        bestSpeedTime = score.bestSpeedTime,
    )

    fun toModel(entity: ScoreEntity) = Score(
        speedScore = entity.speedScore,
        bestSpeedTime = entity.bestSpeedTime,
        totalScore = entity.score
    )
}
